### PR TITLE
Arc - SubclassGenerator needlessly uses field instead of local variable.

### PR DESCRIPTION
**Pull request by manovotn**

### CODE REVIEW

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/SubclassGenerator.java [hunk 1]
```diff
@@ -140,14 +141,9 @@ protected FieldDescriptor createConstructor(ClassOutput classOutput, BeanInfo be
 
         // we build a map for each interceptor instance created, so that they are shared
         // Map<String, Object> where InjectableInterceptor.getIdentifier() is key and Object is instance of the interceptor for this bean
-        FieldCreator interceptorInstanceMap = subclass.getFieldCreator("interceptorInstanceMap", Map.class.getName())
-                .setModifiers(ACC_PRIVATE | ACC_FINAL);
+        AssignableResultHandle interceptorInstanceMap = constructor.createVariable(Map.class);
```
Comment (mkouba) [inline independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/SubclassGenerator.java]: I don't think you need an `AssignableResultHandle` here, `ResultHandle interceptorInstanceMap = constructor.newInstance(MethodDescriptor.ofConstructor(HashMap.class)` should work.

Comment (manovotn) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/SubclassGenerator.java]: Isn't that the same thing? According to Gizmo doc, `constructor.createVariable()` will give me local variable. Or am I missing something?

Comment (mkouba) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/SubclassGenerator.java]: Yes, the output is almost the same. But when using the regular `ResultHandle` a final variable is created and it's just one line - there is no need to invoke `constructor.assign()` afterwards ;-).

Comment (manovotn) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/SubclassGenerator.java]: Ah. got it! Will do :)

**Review by mkouba — APPROVED** [review]

**Pull request merged**